pandas/plotting/_core.py

```diff
@@ -1751,21 +1751,36 @@ def _plot(data, x=None, y=None, subplots=False,
         plot_obj = klass(data, subplots=subplots, ax=ax, kind=kind, **kwds)
     else:
         if isinstance(data, ABCDataFrame):
+            data_cols = data.columns
             if x is not None:
                 if is_integer(x) and not data.columns.holds_integer():
-                    x = data.columns[x]
+                    x = data_cols[x]
                 elif not isinstance(data[x], ABCSeries):
                     raise ValueError("x must be a label or position")
                 data = data.set_index(x)
 
             if y is not None:
-                if is_integer(y) and not data.columns.holds_integer():
-                    y = data.columns[y]
-                elif not isinstance(data[y], ABCSeries):
-                    raise ValueError("y must be a label or position")
-                label = kwds['label'] if 'label' in kwds else y
-                series = data[y].copy()  # Don't modify
-                series.name = label
+                int_y = is_integer(y) or all(is_integer(c) for c in y)
```

Comment:
This looks fragile. For better or worse (mostly worse) you can pretty much any object in pandas columns, e.g. `df = pd.DataFrame({pd.Timestamp('2017'): [1, 2]})`.

I think your code will fail for `y=pd.Timestamp('2017')`, since it's not an integer, but it also isn't iterable.

I'd recommend adding `is_list_like(y)` before the `all`

Reply:
> you can pretty much any object in pandas columns, e.g. df = pd.DataFrame({pd.Timestamp('2017'): [1, 2]})

good call